src/core/random.hpp

```diff
@@ -97,24 +98,28 @@ auto philox_4_uint64s(uint64_t counter, int key1, int key2 = 0) {
  *
  * @tparam salt RNG salt
  * @tparam N    Size of the noise vector
+ * @param counter counter for random number generation
+ * @param seed seed for random number generation
+ * @param key1 key for random number generation
+ * @param key2 key for random number generation
  *
  * @return Vector of uniform random numbers.
  */
 template <RNGSalt salt, size_t N = 3,
           std::enable_if_t<(N > 1) and (N <= 4), int> = 0>
-auto noise_uniform(uint64_t counter, int key1, int key2 = 0) {
+auto noise_uniform(uint64_t counter, uint32_t seed, int key1, int key2 = 0) {
```

Comment:
i think what is missing is an abstraction for retrieving the random numbers specifically for the case of particle ids